pkg/hub/store/store.go

```diff
@@ -74,32 +91,151 @@ func NewClient(cfg *Config) (Client, error) {
+		log.Error(nil, "failed to delete plugins for satellite", zap.String("satellite", satellite), zap.Error(err))
+		return err
+	}
+
+	insert, err := tx.Prepare("INSERT INTO plugins VALUES (?, ?, ?, ?, ?)")
+	if err != nil {
+		return err
+	}
+	defer insert.Close()
+
+	for _, p := range plugins {
+		_, err = insert.Exec(p.Name, p.Type, satellite, p.Description, p.Address)
+		if err != nil {
+			return err
+		}
+	}
+
+	if err = tx.Commit(); err != nil {
+		log.Error(nil, "failed to commit transaction", zap.Error(err))
+		return err
+	}
+
+	return nil
+}
+
+// SaveDashboards deletes all clusters for the given satellite and insert afterwards all passed clusters.
+func (s *client) SaveClusters(satellite string, clusters []string) error {
+	tx, err := s.db.BeginTx(context.Background(), nil)
+	if err != nil {
+		log.Error(nil, "failed to begin transaction", zap.Error(err))
+	}
+	defer tx.Rollback()
+
+	if _, err = tx.Exec("DELETE FROM clusters WHERE satellite=?", satellite); err != nil {
+		log.Error(nil, "failed to delete clusters for satellite", zap.String("satellite", satellite), zap.Error(err))
+		return err
+	}
+
+	insert, err := tx.Prepare("INSERT INTO clusters VALUES (?, ?)")
+	if err != nil {
+		return err
+	}
+	defer insert.Close()
+
+	for _, cluster := range clusters {
+		_, err = insert.Exec(cluster, satellite)
+		if err != nil {
+			return err
+		}
+	}
+
+	if err = tx.Commit(); err != nil {
+		log.Error(nil, "failed to commit transaction", zap.Error(err))
+		return err
+	}
+
+	return nil
+}
+
+// SaveApplications Deletes all applications for the given satellite and insert afterwards all passed applications.
 func (s *client) SaveApplications(satellite string, applications []applicationv1.ApplicationSpec) error {
 	return save(satellite, "application", applications, s)
 }
 
-//SaveDashboards Deletes all dashboards for the given satellite and insert afterwards all passed dashboards.
+// SaveDashboards Deletes all dashboards for the given satellite and insert afterwards all passed dashboards.
 func (s *client) SaveDashboards(satellite string, dashboards []dashboardv1.DashboardSpec) error {
 	return save(satellite, "dashboard", dashboards, s)
 }
 
-//SaveTeams Deletes all teams for the given satellite and insert afterwards all passed teams.
+// SaveTeams Deletes all teams for the given satellite and insert afterwards all passed teams.
 func (s *client) SaveTeams(satellite string, teams []teamv1.TeamSpec) error {
 	return save(satellite, "team", teams, s)
 }
 
-//SaveUsers Deletes all users for the given satellite and insert afterwards all passed users.
+// SaveUsers Deletes all users for the given satellite and insert afterwards all passed users.
 func (s *client) SaveUsers(satellite string, users []userv1.UserSpec) error {
 	return save(satellite, "user", users, s)
 }
 
-//GetApplicationsBySatellite Returns slice of applications for given satellite ordered by name
+func (s *client) GetPlugins(ctx context.Context) ([]plugin.Instance, error) {
```

Comment:
We could offer filtering by satellite, if empty return all.
```suggestion
func (s *client) GetPlugins(ctx context.Context, satellite string) ([]plugin.Instance, error) {
```

Reply:
I think currently, we do not have a use case to get all plugins for a satellite.